torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -786,10 +786,7 @@ std::vector<bool> check_materialization_helper(
         need_materialization.push_back(true);
       }
     } else {
-      // TODO: maybe also handle it is a XLATensor with tensor_data case
-      XLA_CHECK(false)
-          << "_check_tensor_need_materialization "
-             "currently does not handle XLATensor without XLAData and IR";
+      need_materialization.push_back(true);
```

Comment:
I think the right thing to do is to check `xtensor->CurrentTensorData()`, if it is not nullptr then 
```
need_materialization.push_back(false);
```
since we don't need to execute a computation with a XLATensor with `tensor_data`(tensor data is a cpu at::Tensor that we know the value). We should leave the last `else` branch to throw an error to catch any future unhanlded case.

Reply:
Done.